File: Week7/src/models/Product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema(
    {
        name: {
            type: String,
            required: [true, "Tên sản phẩm là bắt buộc"],
        },
        description: {
            type: String,
            default: "",
        },
        price: {
            type: Number,
            required: [true, "Giá sản phẩm là bắt buộc"],
            min: [0, "Giá không được âm"],
        },
        stock: {
            type: Number,
            default: 0,
        },
    },
    { timestamps: true }
);

module.exports = mongoose.model("Product", productSchema);

File: Week7/app/models/product.py
from bson import ObjectId
from datetime import datetime


def product_serializer(product) -> dict:
    """Chuyển MongoDB document thành dict JSON-friendly"""
    return {
        "_id": str(product["_id"]),
        "name": product.get("name", ""),
        "description": product.get("description", ""),
        "price": product.get("price", 0),
        "stock": product.get("stock", 0),
        "createdAt": str(product.get("createdAt", "")),
        "updatedAt": str(product.get("updatedAt", "")),
    }
